### PR TITLE
IRAP 527: Add the cyber "Data transfer" form

### PR DESCRIPTION
**Pull request by Luisella21**

### CODE REVIEW

File: core/forms/irap_request.py [hunk 1]
```diff
@@ -512,6 +512,28 @@ def clean(self):
...
+        if len(cyber_data_transfer_system) > 0 and len(cyber_data_transfer_size) == 0:
+            self.add_error(
+                "data_transfer_size",
+                "is required",
+            )
+        if len(cyber_data_transfer_system) == 0 and len(cyber_data_transfer_size) > 0:
+            self.add_error(
+                "data_transfer_system",
+                "is required",
+            )
```
Comment (SamDudley) [inline core/forms/irap_request.py:533]: I think you may have forgotten to update this validation logic ;)

**Review by SamDudley — CHANGES_REQUESTED** [review]

File: core/forms/irap_request.py (hunk 1 above)
Comment (Luisella21) [reply core/forms/irap_request.py]: I wondered about it.

File: tests/core/test_irap_request.py [hunk 2]
```diff
@@ -720,6 +720,31 @@ def test_submit(self, user):
...
+    def _setup(self, db, client, user):
+        self.client = client
+        self.ir = IrapRequest.objects.create(requestor=user)
+
+    def test_submit(self, user):
+        system_name = "System name"
+        system_syze = "System size"
+        payload = {
+            "data_transfer_system": system_name,
+            "data_transfer_size": system_syze,
```
Comment (SamDudley) [inline tests/core/test_irap_request.py:734]: syze -> size

**Review by SamDudley — CHANGES_REQUESTED** [review]

File: core/forms/irap_request.py [hunk 3]
```diff
@@ -512,6 +512,28 @@ def clean(self):
...
+        fields = ["data_transfer_system", "data_transfer_size"]
+
+    def clean(self):
+        cleaned_data = super().clean()
+        cyber_data_transfer_system = cleaned_data.get("data_transfer_system", "")
+        cyber_data_transfer_size = cleaned_data.get("data_transfer_size", "")
+        if cyber_data_transfer_system and not cyber_data_transfer_size:
+            self.add_error(
+                "data_transfer_size",
+                "is required",
```
Comment (SamDudley) [inline core/forms/irap_request.py:527]: There is a required message in a constant at the top of the file, I think it's called `REQUIRED`.

File: core/forms/irap_request.py [hunk 4]
```diff
@@ -512,6 +512,28 @@ def clean(self):
...
+            )
+        if not cyber_data_transfer_system and cyber_data_transfer_size:
+            self.add_error(
+                "data_transfer_system",
+                "is required",
```
Comment (SamDudley) [inline core/forms/irap_request.py:532]: Same as above.

**Review by SamDudley — CHANGES_REQUESTED** [review]

**Review by SamDudley — APPROVED** [review]

**Pull request merged**